progress-bar-ui.js

```diff
@@ -0,0 +1,337 @@
+        });
+
+        // For whatever reason, Firefox really really doesn't like fill="freeze"
+        // If fill="freeze", Firefox act as if the animation never fully ended
+        // If any of the animationTransition elements' attributes are changed
+        // they will immediately animation instead of waiting for the previous animation to finish
+        // Because fill="freeze" is what makes the effect of the animation stay after it is completed
+        // We need to fake that behaviour
+        // This works but is suuuuuper janky :(
+        if ($.browser.firefox) {
+            $progressBar.find('animateTransform').on('progressend', function() {
+                var $this = $(this);
+                var to = $(this).attr('to');
+
+                $(this).parent().attr('transform', 'rotate(' + to + ')');
+            });
+        }
+
+    };
+
+    // svg defs are global
+    // so, to ensure that each progress bar is clipping correctly, we need to give a unique id to each
+    // unfortunately, this means we have to set a lot of ids :(
+    var _setUniqueIds = function($progressBar) {
+        var id = _uuid();
+
+        $progressBar.data('progressbar-clipid', id);
+
+        // the first four clip paths represent the quadrants
+        // these are the same between all progress bars, so we won't worry about them
+        // the next four are the unique clipping paths
+        // using this weird selector because zepto apparently can't select clippath directly
+        var $uniqueClippingPaths = $progressBar.find('rect').parent().slice(4);
+        var $progressCircles = $progressBar.find('.c-progress-bar__radial-progress');
+
+        $uniqueClippingPaths.each(function(index, clippingPath) {
+            var $clippingPath = $(clippingPath);
+            var clipId = 'progress-bar-' + id + '__clip-' + index;
+
+            $clippingPath.attr('id', clipId);
+            $progressCircles.eq(index).attr('clip-path', 'url(#' + clipId + ')');
+
+            // the animation ids cannot include dashes
+            // if a dash is included in the begin attribute, the animation will not begin
+            // for example, begin="other-anim.end" will never run
+            // but begin="otheranim.end" will run
+            // also, Firefox doesn't like animation ids that start with numbers
+            $clippingPath.find('animateTransform').attr('id', 'anim__' + id + '_' + index);
+        });
+    };
+
+    return {
+        init: function($el, options) {
+            if ($.browser.firefox) {
+                $animation = $el.find('animateTransform');
+                $animation.attr('fill', 'remove');
+                $animation.attr('onend', '$(this).trigger("progressend")');
+            }
+
+            // If not already initialized, create it and expose it through the data method.
+            // Also, expose a separate instance for each progress bar
+            $el.each(function(index, progressBar) {
```

Comment:
As noted, Adaptive will call init for each component instance, so we shouldn’t need to iterate over anything here. AFAIK, we should be able to stick to the pattern outlined in the [Select component](https://github.com/mobify/stencil-select/blob/dev/select-ui.js#L40).